src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.ThreadCounts.cs

```diff
@@ -31,61 +31,52 @@ private struct ThreadCounts : IEquatable<ThreadCounts>
             /// </summary>
             public short NumProcessingWork
             {
-                get => GetInt16Value(NumProcessingWorkShift);
+                get
+                {
+                    short value = GetInt16Value(NumProcessingWorkShift);
+                    Debug.Assert(value >= 0);
+                    return value;
+                }
                 set
                 {
                     Debug.Assert(value >= 0);
-                    SetInt16Value(value, NumProcessingWorkShift);
+                    SetInt16Value(Math.Max((short)0, value), NumProcessingWorkShift);
                 }
             }
 
-            public void SubtractNumProcessingWork(short value)
-            {
-                Debug.Assert(value >= 0);
-                Debug.Assert(value <= NumProcessingWork);
-
-                _data -= (ulong)(ushort)value << NumProcessingWorkShift;
-            }
-
-            public void InterlockedDecrementNumProcessingWork()
-            {
-                Debug.Assert(NumProcessingWorkShift == 0);
-
-                ThreadCounts counts = new ThreadCounts(Interlocked.Decrement(ref _data));
-                Debug.Assert(counts.NumProcessingWork >= 0);
-            }
-
             /// <summary>
             /// Number of thread pool threads that currently exist.
             /// </summary>
             public short NumExistingThreads
             {
-                get => GetInt16Value(NumExistingThreadsShift);
+                get
+                {
+                    short value = GetInt16Value(NumExistingThreadsShift);
+                    Debug.Assert(value >= 0);
+                    return value;
+                }
                 set
                 {
                     Debug.Assert(value >= 0);
-                    SetInt16Value(value, NumExistingThreadsShift);
+                    SetInt16Value(Math.Max((short)0, value), NumExistingThreadsShift);
```

Comment:
can this be changed to `ushort` to avoid the `Math.Max` in multiple places, or is that change going to be too disruptive?

Reply:
Changing to `ushort` wouldn't prevent an underflow, a check would still need to ensure that it does not cross zero and wrap around. It would have other undesirable effects, probably not worth it.